File: backend/src/main/java/com/thanhtam/backend/entity/Course.java
package com.thanhtam.backend.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "course")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Course implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "course_code", nullable = false, unique = true)
    private String courseCode;

    @Column(name = "name")
    private String name;
}

File: source code/front-end-user/src/app/models/course.ts
export class Course {
  id: number;
  courseCode: string;
  name: string;
  imgUrl: string;

  constructor(courseCode: string, name: string, imgUrl: string) {
    this.courseCode = courseCode;
    this.name = name;
    this.imgUrl = imgUrl;
  }

}
